app/main/views/suppliers.py

```diff
@@ -85,13 +86,84 @@ def supplier_details():
     except APIError as e:
         abort(e.status_code)
     supplier['contact'] = supplier['contactInformation'][0]
+    country_name = get_country_name_from_country_code(supplier['registrationCountry'])
 
     return render_template(
         "suppliers/details.html",
         supplier=supplier,
+        country_name=country_name,
     ), 200
 
 
+@main.route('/registered-address/edit', methods=['GET'])
+@login_required
+def edit_registered_address(registered_address_form=None, registered_country_form=None, error=None):
+    try:
+        supplier = data_api_client.get_supplier(
+            current_user.supplier_id
+        )['suppliers']
+    except APIError as e:
+        abort(e.status_code)
+    supplier['contact'] = supplier['contactInformation'][0]
+
+    if not registered_address_form:
+        registered_address_form = EditRegisteredAddressForm(
+            **supplier['contactInformation'][0]
+        )
+
+    if not registered_country_form:
+        registered_country_form = EditRegisteredCountryForm(
+            registrationCountry=supplier['registrationCountry']
+        )
+
+    return render_template(
+        "suppliers/registered_address.html",
+        supplier=supplier,
+        countries=COUNTRY_LIST,
+        registered_address_form=registered_address_form,
+        registered_country_form=registered_country_form,
+        error=error,
+    ), 200
+
+
+@main.route('/registered-address/edit', methods=['POST'])
+@login_required
+def update_registered_address():
+    registered_address_form = EditRegisteredAddressForm()
+    registered_country_form = EditRegisteredCountryForm()
+
+    address_valid = registered_address_form.validate_on_submit()
+    country_valid = registered_country_form.validate_on_submit()
+
+    if not (address_valid and country_valid):
+        return edit_registered_address(
+            registered_address_form=registered_address_form,
+            registered_country_form=registered_country_form,
+        )
+
+    try:
+        data_api_client.update_supplier(
+            current_user.supplier_id,
+            registered_country_form.data,
+            current_user.email_address,
+        )
+
+        data_api_client.update_contact_information(
+            current_user.supplier_id,
+            registered_address_form.id.data,
```

Comment:
Interesting point.

If the GET and POST routes were merged together as you suggest then the Supplier is "got" as the first thing there.  The additional API fetch before posting doesn't feel like a problem, and I'm sure we have that in some other places.

I think you're right it would be nice to avoid leaking these IDs.

Reply:
OK, I've combined the get and post now, and the apiclient call to update contact info is using the id from the supplier. I've also dropped the hidden field in the form in the template which is nice.